### ISSUE TITLE
Widget palette sometimes disappears

### ISSUE THREAD

**Issue by aerwin** [maqetta/maqetta#2728]
I don't have a reliable way to reproduce this, but I often notice the widget palette is empty when an HTML file is open. I think it's often tied to closing files, and then reopening/creating another. But, doesn't always happen.

I've seen this pretty regularly when running from my dev environment. Now, I'm seeing it when running through the test matrix.

May not be much we can do about it until we have a reliable scenario. But, figured we should get it on the books for tracking.

**Comment by JonFerraiolo** [comment]
@jhpedemonte @childsb 

I've seen this too. I looked at the code and didn't see anything wrong after a few minutes of study. However, here are two commits that have happened since M6:
- https://github.com/maqetta/maqetta/commit/297297f709ef419a68048c23d25ec2220db2f31a#diff-6
- https://github.com/maqetta/maqetta/commit/8a4e3dc12340769ec0707eafcd6662a973b8e910#diff-0

**Comment by aerwin** [comment]
Maybe another clue is that much more rarely I see a partial widget palette (almost like vertical position is off):

![screenshot](http://i.imgur.com/MPoxQ.png)

**Comment by JonFerraiolo** [comment]
I think the above problem is different. It has to do with BorderContainer getting confused, particularly when the Chrome debugger is turned on.

But one more thing to mention is that I've seen a second Filter box appear at the bottom of the palette on occasion. So, sometimes we don't create the widget palette when we should (this bug) and sometimes we create a second version when we shouldn't. Note that the logic in Palette.js never destroys the old palette contents. It just has a single property (isLoaded) that sees if the palette needs to be recreated. When it does get recreated, the old content doesn't seem to be purged (or least I haven't found that logic yet).

**Comment by jhpedemonte** [comment]
Dojo 1.8 regression, perhaps?

**Comment by aerwin** [comment]
Ok... think I finally have a reliable recreation scenario:
- New HTML file
- Open Dojo controls section and scroll widget palette all the way to bottom
- Drag text box onto page

![screenshot](http://i.imgur.com/ipoyM.png)
- Save and close
- Reopen HTML file and widget palette not shown:

![screenshot](http://i.imgur.com/HuJZH.png)

I think this is all about vertical positioning... scrolling partially down the widget palette before closing HTML file results in a partially shown widget palette on reopen.
- Partial vertical scroll before closing HTML page:

![screenshot](http://i.imgur.com/0zNiA.png)
- Partial widget palette when reopen page:

![screenshot](http://i.imgur.com/xVeWF.png)

And, in the partial case, opening any of the visible categories causes the whole widget palette to show.

**Comment by jhpedemonte** [comment]
@doronrosenberg, you've been good at figuring out Dojo 1.8 regressions.  Mind taking a quick look at this one?  Might be another such issue.

**Comment by aerwin** [comment]
--> Prio from High to Critical

**Comment by doronrosenberg** [comment]
Windows chrome works fine, mac chrome doesn't.  Firefox doesn't have this issue on any OS.  Hopefully this isn't a browser bug

**Comment by JonFerraiolo** [comment]
> Hopefully this isn't a browser bug

I have seen various scenarios where Chrome's overly aggressive performance optimizations can sometimes cause DOM nodes to render asynchronously, particularly with nested nodes, where DOM nodes don't have `element.ownerDocument.defaultView` in synchronous processing, and instead the browser seems to update rendering and computed style in a separate background thread. There are several spots in our code where we had to workaround such problems, sometimes via a (totally undesirable) setTimeout hack.

But in this case I doubt that's the problem. This is a new regression. We don't have this in M1-M6. It might be Dojo 1.8, but I doubt that too because there isn't much Dojo/Dijit stuff going on with the widgets palette. Instead, the most likely suspect are recent coding changes we made to the widget palette code.
